action/protocol/staking/validations.go

```diff
@@ -39,13 +39,10 @@ func (p *Protocol) validateCreateStake(ctx context.Context, act *action.CreateSt
 	if act.Amount().Cmp(p.config.MinStakeAmount) == -1 {
 		return errors.Wrap(ErrInvalidAmount, "stake amount is less than the minimum requirement")
 	}
-	if act.GasPrice().Sign() < 0 {
-		return errors.Wrap(action.ErrGasPrice, "negative value")
-	}
 	if !p.inMemCandidates.ContainsName(act.Candidate()) {
 		return errors.Wrap(ErrInvalidCanName, "cannot find candidate in candidate center")
 	}
-	return nil
+	return act.AbstractAction.SelfCheck()
```

Comment:
is it right, now actpool will call each incoming selp.Action.Check()?
then we don't need to call it here?